### ISSUE TITLE
algod_ledger_round duplicated in /metrics after running fast catchup

### ISSUE THREAD

**Issue by onetechnical** [algorand/go-algorand#3354]
### Subject of the issue

(As reported by another user) If you run fast catchup, the algod_ledger_round reported by the /metrics endpoint will duplicate stats.

```
# Before Fast Catchup
$ curl -Ss localhost:8080/metrics  | grep -v '#' | grep algod_ledger_round
algod_ledger_round{} 33081

# After Fast Catchup
$ curl -Ss localhost:8080/metrics  | grep -v '#' | grep algod_ledger_round
algod_ledger_round{} 33081
algod_ledger_round{} 18310078

$ curl -Ss localhost:8080/metrics  | grep -v '#' | grep algod_ledger_round
algod_ledger_round{} 33081
algod_ledger_round{} 18310108

$ curl -Ss localhost:8080/metrics  | grep -v '#' | grep algod_ledger_round
algod_ledger_round{} 33081
algod_ledger_round{} 18310142
```

### Your environment

* Software version: 3.2.2

### Steps to reproduce

1. Start a fresh non-archival node
2. Check the /metrics endpoint for `algod_ledger_round`
3. Run fast catchup and complete it
4. Check the /metrics endpoint for `algod_ledger_round` and observe two instances.

### Expected behaviour

Only the latter metric.

### Actual behaviour

Both the former and latter metrics.

A workaround is to restart the algod process.

**Comment by KevinM2k** [comment]
seeing same issue for `algod_ledger_total_transactions`, and `algod_ledger_reward_claims_total` as well as the one mentioned above.

**Comment by 0x4a5e1e4baab** [comment]
also seeing this issue `algod_ledger_transactions_total`, we now have issues in monitoring agent framework due to this duplications.

**Comment by maestroi** [comment]
+1 on above having the same issue

**Comment by macunha1** [comment]
It seems that recent changes to Trackers introduced this bug; all these metrics are getting duplicated in the metrics tracker: https://github.com/algorand/go-algorand/blob/507eb53336959b5cdd6e893ef90828372c7e21d7/ledger/metrics.go#L35-L51

Causing our Prometheus monitoring to throw the following error

```
failed to get prometheus metrics: text format parsing error in line 215: second HELP line for metric name "algod_ledger_transactions_total"
```

which includes the same metrics reported by @onetechnical @0x4a5e1e4baab and @KevinM2k.

```
> curl -s $ALGORAND_HOST/metrics | awk '/algod_ledger/&&!/^#/' | sort

algod_ledger_logic_ok{} 20316
algod_ledger_reward_claims_total{} 123
algod_ledger_reward_claims_total{} 45
algod_ledger_round{} 19142768
algod_ledger_round{} 291
algod_ledger_transactions_total{} 0
algod_ledger_transactions_total{} 1023450
```

@algorandskiy @tsachiherman do you have any idea what may be causing this?

**Comment by algorandskiy** [comment]
What happens here:
1. A node starts, metrics tracker's `loadFromDisk` method creates metrics.
2. Fast catchup completes, ledgers reloads by calling `loadFromDisk` for all trackers.
3. metrics registry ends up with two counters with the same name.

Not clear why it was not a problem before. I do not remember touching metrics logic at all.

**Comment by algorandskiy** [comment]
I did some investigation and looks like we always had such issue. Fixed in #3661
